.tekton/basic-component-pipeline.yaml

```diff
@@ -174,42 +155,42 @@ spec:
       operator: in
       values: [ "true" ]
     workspaces:
-    - name: output
-      workspace: workspace
     - name: basic-auth
       workspace: git-auth
 
   - name: determine-image-tag
     params:
     - name: tag-suffix
       value: $(params.output-tag-suffix)
+    - name: SOURCE_ARTIFACT
+      value: $(tasks.clone-repository.results.SOURCE_ARTIFACT)
     runAfter:
```

Comment:
Getting rid of workspaces makes dependencies between tasks explicit, and, per [this comment](https://redhat-internal.slack.com/archives/C04PZ7H0VA8/p1718876030371379?thread_ts=1718296263.999339&cid=C04PZ7H0VA8), Tekton will be able to infer the pipeline graph based on the usage of params and return values.

This means we can get rid of most `runAfter` things and let Tekton do its smartness.